### PR TITLE
VideoCommon: allow Free Look to be manipulated by a quaternion

### PR DESCRIPTION
**Pull request by iwubcode**

### CODE REVIEW

File: Source/Core/Common/Matrix.cpp [hunk 1]
```diff
@@ -56,6 +56,13 @@ Quaternion Quaternion::RotateZ(float rad)
   return Rotate(rad, Vec3(0, 0, 1));
 }
 
+Quaternion Quaternion::RotateXYZ(const Vec3& rads)
```
Comment (jordan-woyak) [inline Source/Core/Common/Matrix.cpp:59]: There should maybe be a comment that explains this produces a quaternion of simultaneous rotations about the 3 axes. e.g. processing gyroscope input.
I don't know if I like the name `RotateXYZ`, but I don't know what a better name is either. :P

Comment (iwubcode) [reply Source/Core/Common/Matrix.cpp:59]: Yeah, I'm horrible at naming things :D .  Please let me know if you think of something better!

Comment (Rumi-Larry) [reply Source/Core/Common/Matrix.cpp:59]: Here are some suggestions "Rotate_all_axes" "3D_rotation" "Quaternion_rotation"

**Review by leoetlino — CHANGES_REQUESTED** [review]
Changes look sane, but the savestate version needs to be updated.

**Pull request merged**